makeself.sh

```diff
@@ -581,7 +581,7 @@ fi
 tmparch="${TMPDIR:-/tmp}/mkself$$.tar"
 (
     cd "$archdir"
-    find . ! -type d | LC_ALL=C sort | xargs tar $TAR_EXTRA -$TAR_ARGS "$tmparch"
+    find . ! -type d | LC_ALL=C sort | xargs -d '\n' tar $TAR_EXTRA -$TAR_ARGS "$tmparch"
```

Comment:
Unfortunately this doesn't seem to be a standard option for `xargs` - at least the one in macOS doesn't support it, and I bet it's not in the BSD version at all.

Reply:
How about this one:

```suggestion
    find . ! -type d -print0 | LC_ALL=C sort | tr -d '\n' | xargs -0 tar $TAR_EXTRA -$TAR_ARGS "$tmparch"
```

Currently I do not have a server with either macOS or BSD to verify this on those platforms.